fabrics.c

```diff
@@ -721,13 +728,9 @@ static void print_discovery_log(struct nvmf_disc_rsp_page_hdr *log, int numrec,
 		printf("subtype: %s\n", subtype_str(e->subtype));
 		printf("treq:    %s\n", treq_str(e->treq));
 		printf("portid:  %d\n", e->portid);
-		printf("trsvcid: %.*s\n",
-		       space_strip_len(NVMF_TRSVCID_SIZE, e->trsvcid),
-		       e->trsvcid);
+		printf("trsvcid: %.*s\n", NVMF_TRSVCID_SIZE - 1, e->trsvcid);
```

Comment:
This is wrong. We should keep `space_strip_len()` but just make sure we don't do an out of bound access by using `NVMF_TRSVCID_SIZE - 1`.